### DISCUSSION TITLE
Dependency conflict for microprofile-config-api in verison 2.16.x

### DISCUSSION THREAD

**Discussion by iompo** [quarkusio/quarkus#30929] [Q&A] [answered]
Hi all,
In the 2.16.x _quarkus-smallrye-openapi_ depends on _microprofile-config-api_ version 3.0, while most of the other modules depend on version 2.0.1. Everything works fine because of the enforcedPlatform policy, but If I switch from enforcedPlatform to platform the annotation @ConfigProperty stops working. Was that intentional? Because, from my point of view, at least quarkus modules should not depend on different major versions of the same library.

Marco.

**Comment by radcortez** [comment]
This was not intentional. We didn't realize that issue.

Feel free to open an issue to fix it. For reference, here is the original PR: https://github.com/quarkusio/quarkus/pull/29815

**Reply by gsmet** [reply, marked as answer]
While not intentional, Quarkus expects you to use `enforcedPlatform `. You will expose yourself to all sort of problems if you don't do that.
